core/integration/container_test.go

```diff
@@ -4831,3 +4832,74 @@ func (ContainerSuite) TestExecInit(ctx context.Context, t *testctx.T) {
 		require.Contains(t, out, "1 ps")
 	})
 }
+
+func (ContainerSuite) TestContainerAsService(ctx context.Context, t *testctx.T) {
+	c := connect(ctx, t)
+	maingo := `package main
+
+import (
+        "fmt"
+        "net/http"
+        "os"
+				"strings"
+)
+
+func main() {
+        http.HandleFunc("/hello", func(w http.ResponseWriter, r *http.Request) {
+                fmt.Fprintf(w, "args: %s", strings.Join(os.Args, ","))
+        })
+
+        fmt.Println(http.ListenAndServe(":8080", nil))
+}`
+	buildctr := c.Container().
+		From("golang:1.23-alpine").
+		WithWorkdir("/work").
+		WithNewFile("/work/main.go", maingo).
+		WithExec([]string{"go", "build", "-o=app", "main.go"})
+
+	binctr := c.Container().
+		From(alpineImage).
+		WithFile("/bin/app", buildctr.File("/work/app")).
+		WithDefaultArgs([]string{"/bin/app", "via-default-args"}).
+		WithEntrypoint([]string{"/bin/app", "via-entrypoint"}, dagger.ContainerWithEntrypointOpts{KeepDefaultArgs: true}).
```

Comment:
A simple way to avoid the `KeepDefaultArgs` option is to set the entrypoint before the default args.

Reply:
ah. that is good to know. thanks

Reply:
done